metastore/src/main/scala/com/gu/tableversions/metastore/Metastore.scala

```diff
@@ -11,14 +11,16 @@ trait Metastore[F[_]] {
     * Describe the current table in the Metastore interpreted in terms of version information.
     *
     * @param table The table to query
+    * @return the table version information, or a failure if the table doesn't exist or can't be queried.
     */
-  def currentVersion(table: TableName): F[Option[TableVersion]]
+  def currentVersion(table: TableName): F[TableVersion]
```

Comment:
I changed the return type for this back to `F[TableVersion]` as I found that a) it was confusing what `None` meant, and b) all caller code wanted to treat this as an error anyway so had to explicitly turn this into a failure.